apollo-api/src/main/java/com/apollographql/apollo/api/internal/json/BufferedSourceJsonReader.kt

```diff
@@ -55,7 +55,7 @@ class BufferedSourceJsonReader(private val source: BufferedSource) : JsonReader
   override var failOnUnknown = false
 
   @Throws(IOException::class)
-  override fun beginArray() {
+  override fun beginArray(): JsonReader {
```

Comment:
One thing that I do for self returning functions is to use apply

```
override fun beginArray(): JsonReader = apply {
  // do stuff
}
```

Then you don't have to `return this`